sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/TransactionalBatchResponse.java

```diff
@@ -0,0 +1,340 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.JsonSerializable;
+import com.azure.cosmos.implementation.apachecommons.collections.list.UnmodifiableList;
+import com.azure.cosmos.implementation.batch.ItemBatchOperation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+/**
+ * Response of a {@link TransactionalBatch} request.
+ */
+public class TransactionalBatchResponse implements AutoCloseable, List<TransactionalBatchOperationResult<?>> {
+
+    private final static Logger logger = LoggerFactory.getLogger(TransactionalBatchResponse.class);
+
+    private Map<String, String> responseHeaders;
+    private final int responseStatus;
+    private String errorMessage;
+    private List<TransactionalBatchOperationResult<?>> results;
+    private int subStatusCode;
+    private List<ItemBatchOperation<?>> operations;
+    private CosmosDiagnostics cosmosDiagnostics;
+
+    /**
+     * Initializes a new instance of the {@link TransactionalBatchResponse} class.
+     *
+     * @param responseStatus the  response status.
+     * @param subStatusCode the response sub-status code.
+     * @param errorMessage an error message or {@code null}.
+     * @param responseHeaders the response http headers
+     * @param cosmosDiagnostics the diagnostic
+     * @param operations a {@link List list} of {@link ItemBatchOperation batch operations}.
+     */
+    public TransactionalBatchResponse(
+        final int responseStatus,
+        final int subStatusCode,
+        final String errorMessage,
+        final Map<String, String> responseHeaders,
+        final CosmosDiagnostics cosmosDiagnostics,
+        final List<ItemBatchOperation<?>> operations) {
```

Comment:
This leaks an implementation class into the public surface area. Either ItemBatchOperation needs to be moved to public area (com.azure.cosmos.model) or not be exposed here.

Reply:
This can be required for bulk with current implementation, but ItemBatchOperation is how we keep operation In TransactionalBatch. May be it can be moved in Model then.

Reply:
But if the get functions are package private in TransactionalBatchResponse and TransactionalBatch, then is it needed to keep it in public ?

Reply:
All good now - as long as it is only exposed in package private API it isn't a problem

Reply:
Cool. Thanks!!